docs/guides/using_TransformAnimationComponent.rst

```diff
@@ -0,0 +1,160 @@
+===================================================
+Animating Objects using TransformAnimationComponent
+===================================================
+
+TransformAnimationComponent can be used to easily animate scenes which have to be animated endlessly without relying on scripting for the same.
+
+It is quite easy to as it uses a keyframe based interface to achieve this.
+
+Setting up
+----------
+
+Create a new scene and assign the TransformComponent, ModelComponent and TransformAnimationComponent to it.
+
+Creating a new scene:
+
+.. image:: images/AnimTestCreate.png
+
+Adding the Components by right clicking on the scene in inspector:
+
+.. image:: images/AnimTestComp.png
+
+Select the EditorCamera:
+
+.. image:: images/AnimTestCam.png
+
+Set the camera to EditorCamera and position it such that it has a good view of the scene.
+
+.. image:: images/AnimTestView.png
+
+Select the EditorCamera as default camera for the scene.
+
+.. image:: images/AnimTestDefCam.png
+
+Interface
+---------
+
+* Transition Type : How the transition must be between keyframes. Explaining the options:
+
+  - SmashSmash : Abrupt start and end transition
+  - EaseEase : Smooth start and end transition
+  - SmashEase : Abrubt start and smooth end transition
+  - EaseSmash : Smooth start and abrupt end transition
+
+* Animation Mode : How the animation should play. Explaining the options:
+
+  - Looping : Animation plays from Start time to End time unidirectionally.
+  - Alternating : Animation plays from Start time to EndTime and then reverses from End time to Start time.
```

Comment:
The nesting here isn't working as intended. Also, instead of explaining the options, maybe write something like "Following Transition Types/Animation Modes are possible"
![image](https://user-images.githubusercontent.com/54415525/155853171-29d6dc8c-e52e-49b2-bc96-8ccc39e1503e.png)